File: algorithms/dynamic-programming/rod-cutting-algorithm/python/rod_cutting.py
def rod_cut(prices, n):
    dp = [0] * (n + 1)

    for i in range(1, n + 1):
        for j in range(i):
            dp[i] = max(dp[i], prices[j] + dp[i - j - 1])

    return dp[n]


if __name__ == "__main__":
    prices = [1, 5, 8, 9, 10, 17, 17, 20]
    print(rod_cut(prices, 8))  # 22

File: algorithms/dynamic-programming/rod-cutting-algorithm/swift/RodCutting.swift
func rodCut(_ prices: [Int], _ n: Int) -> Int {
    var dp = Array(repeating: 0, count: n + 1)

    if n > 0 {
        for i in 1...n {
            for j in 0..<i {
                dp[i] = max(dp[i], prices[j] + dp[i - j - 1])
            }
        }
    }

    return dp[n]
}

print(rodCut([1, 5, 8, 9, 10, 17, 17, 20], 8)) // 22
